test/Microsoft.ApplicationInsights.AspNet.Tests/TelemetryInitializers/OperationNameTelemetryInitializerTests.cs

```diff
@@ -72,16 +70,107 @@ public void InitializeSetsTelemetryOperationNameToMethodAndPath()
+
+            var telemetry = new EventTelemetry();
+            initializer.Initialize(telemetry);
+
+            Assert.Equal("GET account/login", telemetry.Context.Operation.Name);
+        }
+
+        [Fact]
+        public void InitializeSetsTelemetryOperationNameToControllerAndActionAndParameterFromActionContext()
+        {
+            var actionContext = new ActionContext();
+            actionContext.RouteData = new RouteData();
+            actionContext.RouteData.Values.Add("controller", "account");
+            actionContext.RouteData.Values.Add("action", "login");
+            actionContext.RouteData.Values.Add("parameter", "myName");
+
+            var contextAccessor = HttpContextAccessorHelper.CreateHttpContextAccessor(new RequestTelemetry(), actionContext);
+
+            var initializer = new OperationNameTelemetryInitializer(contextAccessor);
+
+            var telemetry = new EventTelemetry();
+            initializer.Initialize(telemetry);
+
+            Assert.Equal("GET account/login [parameter]", telemetry.Context.Operation.Name);
+        }
+
+        [Fact]
+        public void InitializeSortsParameters()
+        {
+            var actionContext = new ActionContext();
+            actionContext.RouteData = new RouteData();
+            actionContext.RouteData.Values.Add("controller", "account");
+            actionContext.RouteData.Values.Add("action", "login");
+            actionContext.RouteData.Values.Add("parameterZ", "myName1");
+            actionContext.RouteData.Values.Add("parameterA", "myName2");
+            actionContext.RouteData.Values.Add("parameterN", "myName1");
+
+            var contextAccessor = HttpContextAccessorHelper.CreateHttpContextAccessor(new RequestTelemetry(), actionContext);
+
+            var initializer = new OperationNameTelemetryInitializer(contextAccessor);
+
+            var telemetry = new EventTelemetry();
+            initializer.Initialize(telemetry);
+
+            Assert.Equal("GET account/login [parameterA/parameterN/parameterZ]", telemetry.Context.Operation.Name);
+        }
+
+        [Fact]
+        public void InitializeDoesNotIncludeRouteGroupKeyInParametersList()
+        {
+            var actionContext = new ActionContext();
+            actionContext.RouteData = new RouteData();
+            actionContext.RouteData.Values.Add("controller", "account");
+            actionContext.RouteData.Values.Add("action", "login");
+            actionContext.RouteData.Values.Add(AttributeRouting.RouteGroupKey, "RouteGroupKey");
+            
+            var contextAccessor = HttpContextAccessorHelper.CreateHttpContextAccessor(new RequestTelemetry(), actionContext);
+
+            var initializer = new OperationNameTelemetryInitializer(contextAccessor);
+
+            var telemetry = new EventTelemetry();
+            initializer.Initialize(telemetry);
+
+            Assert.Equal("GET account/login", telemetry.Context.Operation.Name);
```

Comment:
Assert is not that complex to try to simplify it but we also check for any other side effects doing Equal